src/webui/api/searchcontroller.cpp

```diff
@@ -0,0 +1,312 @@
+    const QString category = params()["category"].trimmed();
+    const QString plugins = params()["plugins"].trimmed();
+    const QString pluginsLower = plugins.toLower();
+
+    QStringList pluginsToUse;
+    if (pluginsLower == "all") pluginsToUse = SearchPluginManager::instance()->allPlugins();
+    else if (pluginsLower == "enabled") pluginsToUse = SearchPluginManager::instance()->enabledPlugins();
+    else if (pluginsLower == "multi") pluginsToUse = SearchPluginManager::instance()->enabledPlugins();
+    else pluginsToUse << plugins;
+
+    const QString id = generateSearchId();
+    m_searches[id] = SearchPluginManager::instance()->startSearch(pattern, category, pluginsToUse, true);
+
+    const QJsonObject result = {
+        {"status", "Ok"},
+        {"id", id}
+    };
+    setResult(result);
+}
+
+void SearchController::stopAction()
+{
+    checkParams({"id"});
+
+    const QString id = params()["id"].trimmed();
+    if (!m_searches.contains(id))
+        throw APIError(APIErrorType::BadParams);
+
+    SearchHandler *searchHandler = m_searches[id];
+
+    if (searchHandler->isActive())
+        searchHandler->cancelSearch();
+
+    const QJsonObject result = {
+        {"status", "Ok"}
+    };
+    setResult(result);
+}
+
+void SearchController::statusAction()
+{
+    checkParams({"id"});
+
+    const QString id = params()["id"].trimmed();
+    if (!m_searches.contains(id))
+        throw APIError(APIErrorType::BadParams);
+
+    SearchHandler *searchHandler = m_searches[id];
+
+    const QJsonObject result = {
+        {"status", "Ok"},
+        {"result", searchHandler->isActive() ? "Running" : "Stopped"}
+    };
+    setResult(result);
+}
+
+void SearchController::resultsAction()
+{
+    checkParams({"id"});
+
+    int max = INT_MAX;
+    bool ok = true;
+    const QString id = params()["id"].trimmed();
+    const QString maxStr = params()["max"].trimmed();
+    if (!maxStr.isEmpty())
+        max = maxStr.toInt(&ok);
+
+    if (!ok || !m_searches.contains(id))
+        throw APIError(APIErrorType::BadParams);
+
+    SearchHandler *searchHandler = m_searches[id];
+    const QList<SearchResult> searchResults = searchHandler->queuedResults(max);
+
+    setResult(getResults(searchResults, searchHandler->isActive(), searchHandler->getQueueSize()));
+
+    if (!searchHandler->isActive() && (searchHandler->getQueueSize() == 0))
+        m_searches.remove(id);
+}
+
+void SearchController::deleteAction()
+{
+    checkParams({"id"});
+
+    const QString id = params()["id"].trimmed();
+    if (!m_searches.contains(id))
+        throw APIError(APIErrorType::BadParams);
+
+    SearchHandler *searchHandler = m_searches[id];
+    searchHandler->cancelSearch();
+    m_searches.remove(id);
+
+    const QJsonObject result = {
+        {"status", "Ok"}
+    };
+    setResult(result);
+}
+
+void SearchController::categoriesAction()
+{
+    QStringList categories;
+    const QString name = params()["name"].trimmed();
```

Comment:
The query like "search/categories?name=somename" is confusing. What does name mean? Shouldn't it be "pluginName"?